### PR TITLE
Improve Contributor experience

### PR DESCRIPTION
**Pull request by TWiStErRob**

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Contribution Guidelines
+
+## IDE
+
+IntelliJ IDEA Community / Ultimate. (Please report any issues with Community as I'm using Ultimate.)
+
+Recommended plugins:
+
+* detekt (config file: [config/detekt/detekt.yml](../config/detekt/detekt.yml))
```
Comment (timsearle) [inline docs/CONTRIBUTING.md:9]:
```suggestion
* [detekt](https://plugins.jetbrains.com/plugin/10761-detekt) (config file: [config/detekt/detekt.yml](../config/detekt/detekt.yml))
```

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  For a faster feedback loop.
+* Markdown
```
Comment (timsearle) [inline docs/CONTRIBUTING.md:11]:
```suggestion
* [Markdown](https://plugins.jetbrains.com/plugin/7793-markdown)
```

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+Notable things:
+
+* The project uses TABs as indent, except for YAML.
+* CI / `gradlew build` will tell you off for most things, so reviews don't have to.
+* The `.idea` folder is version controlled, this means when you open the project it should pick up:
+  * Code styles specific to the project
+  * Inspection profile specific to the project
+
+  With these, it is recommended to write code without any warnings.
+  Use _menu > Code > Reformat Code_ when editing files, or turn on _Commit Checks > Reformat Code_ in the commit dialog.
```
Comment (timsearle) [inline docs/CONTRIBUTING.md:29]:
```suggestion
  Use menu _Code > Reformat Code_ when editing files, or turn on _Commit Checks > Reformat Code_ in the commit dialog.
```

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Notable things:
+
+* The project uses TABs as indent, except for YAML.
+* CI / `gradlew build` will tell you off for most things, so reviews don't have to.
+* The `.idea` folder is version controlled, this means when you open the project it should pick up:
+  * Code styles specific to the project
+  * Inspection profile specific to the project
+
+  With these, it is recommended to write code without any warnings.
```
Comment (timsearle) [inline docs/CONTRIBUTING.md:28]: Isn't `-Werror` specified? So not really a "recommendation"?

```
e: warnings found and -Werror specified
w: file:///Users/timsearle/Developer/open-source/net.twisterrob.ghlint/modules/ghlint-rules/src/test/kotlin/net/twisterrob/ghlint/rules/ExplicitJobPermissionsRuleTest.kt:259:8 Variable 'unused' is never used
```

Comment (TWiStErRob) [reply docs/CONTRIBUTING.md:28]: I tried running IDEA on CI, but it was too slow / reporting required a third party.

Comment (TWiStErRob) [reply docs/CONTRIBUTING.md:28]: I'll see if I can raise severity in IDEA for these for a tighter loop.

Comment (TWiStErRob) [reply docs/CONTRIBUTING.md:28]: Well, no. https://youtrack.jetbrains.com/issue/KTIJ-16488

File: modules/ghlint-reporting-sarif/src/main/kotlin/net/twisterrob/ghlint/reporting/sarif/SarifReporter.kt [hunk 5]
```diff
@@ -100,7 +100,7 @@ private fun reportingDescriptor(issue: Issue): ReportingDescriptor =
 	)
 
 private fun result(finding: Finding, base: Path): Result {
-	val file = Path.of(finding.location.file.path).absolute()
+	val file = Path.of(finding.location.file.path).toRealPath(LinkOption.NOFOLLOW_LINKS)
```
Comment (TWiStErRob) [inline modules/ghlint-reporting-sarif/src/main/kotlin/net/twisterrob/ghlint/reporting/sarif/SarifReporter.kt:103]: @timsearle this is the fix.

Can you please run `gradlew :ghlint-reporting-sarif:build` to verify locally?

Comment (timsearle) [reply modules/ghlint-reporting-sarif/src/main/kotlin/net/twisterrob/ghlint/reporting/sarif/SarifReporter.kt:103]: Screenshot 2024-06-02 at 17 35 04
Screenshot 2024-06-02 at 17 34 34

**Pull request merged**